index.ts

```diff
@@ -200,16 +204,22 @@ cron.onEvent("partition-registrar", new CallbackFunction('partition-callback', {
     
         const client = athena.createClient(clientConfig, awsConfig);
     
-        const date = moment(event.time).utc().format("YYYY/MM/DD/HH");
-    
-        const query = `ALTER TABLE ${db.name.get()}.logs add
-        PARTITION (inserted_at = '${date}') LOCATION '${location.get()}/${date}/';`;
+        let date = moment(event.time);
+        
+        for(let i = 0; i <= 12; i++) {
+            const dateString = date.utc().format("YYYY/MM/DD/HH");
+
+            const query = `ALTER TABLE ${db.name.get()}.logs ADD IF NOT EXISTS
+            PARTITION (inserted_at = '${dateString}') LOCATION '${location.get()}/${dateString}/';`;
     
-        client.execute(query, (err: Error) => {
-            if (err) {
-                throw err;
-            }
-        })
+            client.execute(query, (err: Error) => {
```

Comment:
Is this an async api? Does this mean that we'll be issuing 12 concurrent queries? [Athena service limits](https://docs.aws.amazon.com/athena/latest/ug/service-limits.html) indicate a max of five concurrent queries. It may be better to build up a single query string that registers all 12 partitions in one statement. See the last example [here](https://docs.aws.amazon.com/athena/latest/ug/alter-table-add-partition.html):

```sql
ALTER TABLE orders ADD
  PARTITION (dt = '2016-05-14', country = 'IN') LOCATION 's3://mystorage/path/to/INDIA_14_May_2016/'
  PARTITION (dt = '2016-05-15', country = 'IN') LOCATION 's3://mystorage/path/to/INDIA_15_May_2016/';
```

Reply:
Will now only use a single query that uses method in partitionHandler